pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -107,6 +119,8 @@ rec {
             ++ optional (elem "host"   configurePlatforms) "--host=${stdenv.hostPlatform.config}"
             ++ optional (elem "target" configurePlatforms) "--target=${stdenv.targetPlatform.config}";
 
+        } // lib.optionalAttrs (hardeningDisable != [] || hardeningEnable != []) {
+          NIX_HARDENING_ENABLE = enabledHardeningOptions;
```

Comment:
@Ericson2314 Regarding your comment [regarding `NIX_CC_WRAPPER_@infixSalt@_HARDENING_ENABLE`](https://github.com/NixOS/nixpkgs/pull/28029/files#r133272454), what's the appropriate thing to do here? Should I mostly leave this alone (the only change, perhaps, being a rename to `NIX_CC_WRAPPER_HARDENING_ENABLE`), and then have (per another one of your recommendations) the setupHook for the CC-wrapper use the `$NIX_CC_WRAPPER_HARDENING_ENABLE` when `NIX_CC_WRAPPER_@infixSalt@_HARDENING_ENABLE` isn't defined? Curious what the exact logic should be here.

Reply:
1. https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/cc-wrapper/add-flags.sh#L7-L13 in there goes `NIX_CC_WRAPPER+HARDENING_ENABLE`.
2. this line becomes `NIX_CC_WRAPPER_HARDENING_ENABLE ..`

Now that there's a bintools wrapper too, we would have to do an analogous thing there with this plan. So OTOH if we want just one variable with both, then maybe we shouldn't add the `_CC_WRAPPER` to the name. *But*, no matter what we still want to do the infix salt.